astronomer/providers/sftp/sensors/sftp.py

```diff
@@ -43,18 +42,17 @@ def execute(self, context: Context) -> None:
         Logic that the sensor uses to correctly identify which trigger to
         execute, and defer execution as expected.
         """
-        if not self.poke(context=context):
```

Comment:
Why did you remove the sync check before deferring? Probably worth adding a comment saying why we don't follow that pattern for this Sensor

Reply:
I have added a comment describing the reason. Please check.